lib/rules/UseEffectRulePlugin.js

```diff
@@ -27,27 +63,33 @@ class UseEffectRulePlugin {
 					}
 				};
 
+				unhandledProperties.delete("enforce");
+
+				/** @type {string} */
+				const type = rule.enforce ? `use-${rule.enforce}` : "use";
+
 				if (unhandledProperties.has("use")) {
 					unhandledProperties.delete("use");
-					unhandledProperties.delete("enforce");
 
 					conflictWith("loader", "use");
 					conflictWith("options", "use");
 
+					/** @typedef {RuleSetUse} */
 					const use = rule.use;
-					const enforce = rule.enforce;
-
-					const type = enforce ? `use-${enforce}` : "use";
 
 					/**
-					 *
 					 * @param {string} defaultIdent default ident when none is provided
-					 * @param {object} item user provided use value
+					 * @param {UseDefinition} item user provided use value
 					 * @returns {Effect|function(any): Effect[]} effect
 					 */
 					const useToEffect = (defaultIdent, item) => {
 						if (typeof item === "function") {
-							return data => useToEffectsWithoutIdent(item(data));
+							return data =>
+								useToEffectsWithoutIdent(
+									/** @type {string|UseRule|Array<string|UseRule>} */ (item(
```

Comment:
seems like `useToEffectsWithoutIdent` does not accept `RuleSetUseItem` as function, but it is presented in definition https://github.com/webpack/webpack/blob/next/declarations/WebpackOptions.d.ts#L193